src/tools/morpho/index.ts

```diff
@@ -1,10 +1,44 @@
-import { generateTool } from '../../utils.js';
-import { getMorphoVaultsHandler } from './handlers.js';
+import {
+  ActionProvider,
+  CreateAction,
+  type EvmWalletProvider,
```

Comment:
importing the class only as type removes the runtime reflection metadata we rely on to inspect the tool args.